hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/prehandle/PreHandleContextImpl.java

```diff
@@ -353,18 +363,21 @@ public PreHandleContext requireKeyIfReceiverSigRequired(
         if (!account.receiverSigRequired()) {
             return this;
         }
-
+        // If it is hollow account, and we require this to sign, we need to finalize the account
```

Comment:
It would be nice to collapse as much as possible of `requireKeyOrThrow ` and `requireKeyIfReceiverSigRequired` into a common method so we don't have to duplicate any of the logic across both methods, but not a blocker for this PR

Reply:
Agreed. Could not find a way that time.